sdk/tests/deqp/modules/shared/glsShaderExecUtil.js

```diff
@@ -685,10 +685,11 @@ goog.scope(function() {
                 gl.readBuffer(gl.COLOR_ATTACHMENT0 + outLocation + locNdx);
                 gl.readPixels(0, 0, framebufferW, framebufferH, transferFormat.format, transferFormat.dataType, tmpBuf.getAccess().getDataPtr());
 
-                if (outSize == 4 && outNumLocs == 1)
+                if (outSize == 4 && outNumLocs == 1) {
                     outputs[outNdx] = new Uint8Array(tmpBuf.getAccess().getBuffer());
-                else {
-                    outputs[outNdx] = new Uint32Array(numValues * outVecSize);
+                } else {
+                    if (locNdx == 0)
+                        outputs[outNdx] = new Uint32Array(numValues * outVecSize);
```

Comment:
The inner 'for' loop from line 684 to line 700 will insert data to the same array by setting offset during each iteration. We just need to allocate one array for each output (It is the result of one 'out' variable in fragment shader) at the beginning. If we allocate multiple arrays in the inner 'for' loop, data stored in arrays (except the last one) are lost.